src/http/log.cr

```diff
@@ -0,0 +1,19 @@
+require "log"
+
+class HTTP::Client
+  Log = ::Log.for(self)
+
+  protected def before_exec(request)
```

Comment:
These aren't documented. How are people going to find out they can do this?

Reply:
It's harder to write English than crystal :-) I'll add documentation after discussing if this is the right approach